### PR TITLE
fix: upgrade http-foundation to 5.0

### PR DESCRIPTION
**Pull request by muchafm**
Hi :) 
This is my first OSS contribution :)
So this is a small fix due to a security alert, I just upgrade the http-foundation lib => https://github.com/KnpLabs/KnpMenu/network/alert/composer.json/symfony%2Fhttp-foundation/open

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -28,7 +28,7 @@
     "require-dev": {
         "phpspec/prophecy": "^1.8",
         "psr/container": "^1.0",
-        "symfony/http-foundation": "^3.4 || ^4.2|| ^5.0",
+        "symfony/http-foundation": "^5.1",
```
Comment (garak) [inline composer.json:31]: We can't exclude versions 3 and 4 of Symfony

**Review by garak — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (muchafm) [reply composer.json:31]: oh, I see ! I will modify this, thank !

**Comment by derrabus** [comment]
I don't really see what you want to achieve with this change. This library is not suddenly incompatible with HttpFoundation 5.0.

**Comment by stof** [comment]
the handling of security alerts on github for the composer ecosystem is a buggy in my opinion. they apply the same logic to projects locking their dependencies and to libraries not shipping any lock file. Finding vulnerable dependencies makes sense when you know the version being used. But currently, their alert system expects all libraries to update all their constraints to allow only non-vulnerable versions.
I think this is inherited from the npm ecosystem, where npm can install separate versions in the dependency tree and does not guarantee that a deep part of the tree gets upgraded if there is no new version of the parent package in the tree. But that's not how composer updates are working.

**Comment by garak** [comment]
I guess that we could drop support for Symfony 5.0, being [out of support](https://symfony.com/releases/5.0).
If so, for consistence, we should also drop support for Symfony 4.2 and 4.3

**Comment by muchafm** [comment]
@stof made a good point, I'm just going to close this pr :)

**Pull request closed without merging**